swarm/src/lib.rs

```diff
@@ -1901,10 +1914,14 @@ mod tests {
     fn test_swarm_disconnect() {
         // Since the test does not try to open any substreams, we can
         // use the dummy protocols handler.
-        let handler_proto = keep_alive::ConnectionHandler;
+        let handler_proto = dummy::ConnectionHandler;
 
-        let mut swarm1 = new_test_swarm::<_, ()>(handler_proto.clone()).build();
-        let mut swarm2 = new_test_swarm::<_, ()>(handler_proto).build();
+        let mut swarm1 = new_test_swarm::<_, ()>(handler_proto.clone())
+            .idle_connection_timeout(Duration::from_secs(5))
```

Comment:
If we end up configuring this in every test, we might as well do it inside `new_test_swarm`!